types/react-dom/client.d.ts

```diff
@@ -33,10 +31,10 @@ export interface Root {
  *
  * @see https://reactjs.org/docs/concurrent-mode-reference.html#createroot
  */
-export function createRoot(container: Element | Document | DocumentFragment | Comment, options?: RootOptions): Root;
+export function createRoot(container: Element | DocumentFragment, options?: RootOptions): Root;
 
 export function hydrateRoot(
-    container: Element | Document | DocumentFragment | Comment,
+    container: Element | Document | DocumentFragment,
```

Comment:
Oh actually, this shouldn't have DocumentFragment. Mostly because I'm not sure it works but it might.

Reply:
See https://github.com/facebook/react/pull/24110

Reply:
Right, makes sense.
```suggestion
    container: Element | Document,
```